actualize Vagrant and fix problems on orchestrator-agent api with sqlite backend

Slach opened the pull request:
Related issue: https://github.com/github/orchestrator/issues/438

### Description

This PR actualize Vagrant and fix problems on orchestrator-agent api with sqlite backend
i hope this help use innodbbackupex for data seeding between two mysql nodes

> In case this PR introduced Go code changes:

- [x] contributed code is using same conventions as original code
- [x] code is formatted via `gofmt` (please avoid `goimports`)
- [x] code is built via `./build.sh`
- [x] code is tested via `go test ./go/...`

Slach committed:
actualize Vagrant environment and fix problems on orchestrator-agent api with sqlite backend

Signed-off-by: Slach <bloodjazman@gmail.com>

Slach committed:
add sourceHost parameter into agent /api/post-copy/

shlomi-noach commented on go/agent/agent_dao.go:
```diff
@@ -547,8 +547,8 @@ func AbortSeed(seedId int64) error {
 // PostCopy will request an agent to invoke post-copy commands
-func PostCopy(hostname string) (Agent, error) {
-	return executeAgentCommand(hostname, "post-copy", nil)
+func PostCopy(hostname, sourceHostname string) (Agent, error) {
```
Is this change going to break behavior for existing users?

Slach replied:
could you help me figure out howto add optional parameter to martini routes?

shlomi-noach replied:
Sure. Consider, for example,

https://github.com/github/orchestrator/blob/f465c53809370697566ca84b418fd43f7fe41ca9/go/http/api.go#L3291-L3292

and then

https://github.com/github/orchestrator/blob/f465c53809370697566ca84b418fd43f7fe41ca9/go/http/api.go#L2696-L2726

Slach replied:
@shlomi-noach ok. 
i add backwards compatibility on orchestrator-agent here
https://github.com/github/orchestrator-agent/pull/20/files#diff-fbdc8cde28a71961e53c2e83e2483c9aL619

and add some other improvments and tested it with percona-xtrabackup
currenly in my setup i have successfull data seed between nodes over netcat + xtrabackup + xbstream

Slach committed:
Merge branch 'master' of github.com:github/orchestrator into reanimate_agent

Slach committed:
sourceHostname to logging

Slach committed:
add include /etc/ files for iject DAEMONOPTS or other settings before daemon running

Slach committed:
more clarified log message

Slach committed:
add SeedWaitSecondsBeforeSend config parameter

shlomi-noach commented on etc/init.d/orchestrator.bash:
```diff
@@ -54,8 +54,9 @@ start_daemon () {
-# The file /etc/orchestrator_profile can be used to inject pre-service execution
+# This files can be used to inject pre-service execution
 # scripts, such as exporting variables or whatever. It's yours!
+[ -f /etc/default/orchestrator ] && . /etc/default/orchestrator
```
Please explain why `/etc/default/orchestrator`? IF anything, I'd add `/etc/profile.d/orchestrator` which is what I should have done in the first place.

Slach replied:
/etc/default it's a default behavion on debian based distributives

Slach replied:
ok, i add usage  ```/etc/profile.d/orchestrator``` and ```/etc/default/orchestrator``` 

```/etc/profile.d/orchestrator``` will have major priority

shlomi-noach commented:
Thank you! This PR is growing bigger and deals with different problems, so it's becoming difficult for me to review it as a whole. Can you please elaborate on "i add backwards compatibility on orchestrator-agent here" ? Where is the backwards compatibility found?

Slach committed:
backwards compatibility for api/post-copy agent command

Slach commented on go/agent/agent_dao.go:
```diff
@@ -164,7 +164,7 @@ func ReadOutdatedAgentsHosts() ([]string, error) {
 			host_agent
 		where
-			IFNULL(last_checked < now() - interval ? minute, true)
+			IFNULL(last_checked < now() - interval ? minute, 1)
```
this change need for SQLlite compatibility cause SQLite desn't have BOOLEAN literal

Slach commented on go/agent/agent_dao.go:
```diff
@@ -547,8 +547,8 @@ func AbortSeed(seedId int64) error {
-func PostCopy(hostname string) (Agent, error) {
-	return executeAgentCommand(hostname, "post-copy", nil)
+func PostCopy(hostname, sourceHostname string) (Agent, error) {
+	return executeAgentCommand(hostname, fmt.Sprintf("post-copy/?sourceHost=%s", sourceHostname), nil)
```
there is backwards compatibility

Slach commented:
>Can you please elaborate on "i add backwards compatibility on orchestrator-agent here" ?
backwards behavior it's here

https://github.com/github/orchestrator-agent/pull/20/files#diff-fbdc8cde28a71961e53c2e83e2483c9aL619
and here
https://github.com/github/orchestrator/pull/445/files#diff-17473f4f3d0788c153babef6ddbba1faR551

Slach committed:
add /etc/profile.d/orchestrator usage

shlomi-noach commented:
I will get back on this early next week.

Slach commented:
ok. i hope my PR will be merged ;)

shlomi-noach commented on .gitignore:
```diff
@@ -13,3 +14,7 @@ vagrant/vagrant-ssh-key.pub
+.idea/
+*.deb
+*.pcap
+*.log
```
:+1:

shlomi-noach commented on Vagrantfile:
```diff
@@ -16,8 +16,8 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
-  config.vm.synced_folder '.', '/orchestrator', type: 'rsync',
-    rsync__auto: true
+  config.vm.synced_folder '.', '/orchestrator'
+  #, type: 'rsync', rsync__auto: true
```
Can you please explain this change?

Slach replied:
rsync__auto want work on my Vagrant environment ;(
when i change some sources on host OS, it's not rsynced into guest maching
ok. i rollback this change

shlomi-noach replied:
Sorry, please elaborate. Did you introduce `rsync__auto` in the first place?

Slach replied:
i try use following variants
1)
```
 config.vm.synced_folder '.', '/orchestrator', type: 'rsync'
```

it's require run ```vagrant reload``` when i change any file on my host machine
2)
```
 config.vm.synced_folder '.', '/orchestrator', type: 'rsync',  rsync__auto: true
```
it not worked, and i don't investigate why
then i just commented this options and default sharing mode over vboxfs works as i expected

shlomi-noach commented on build.sh:
```diff
@@ -87,7 +87,7 @@ function oinstall() {
   gofmt -s -w  go/
   rsync -qa ./resources $builddir/orchestrator${prefix}/orchestrator/
-  rsync -qa ./conf/orchestrator-sample.* $builddir/orchestrator${prefix}/orchestrator/
+  rsync -qa ./conf/orchestrator-sample*.conf.json $builddir/orchestrator${prefix}/orchestrator/
```
:+1:

shlomi-noach commented on build.sh:
```diff
@@ -134,12 +134,14 @@ function package() {
-    # n CentOD 6 box: we only want the rpms for CentOS6
-    # Add "-centos6" to the file name.
-    ls ${TOPDIR:-?}/*.rpm | while read f; do centos_file=$(echo $f | sed -r -e "s/^(.*)-${RELEASE_VERSION}(.*)/\1-centos6-${RELEASE_VERSION}\2/g") ; mv $f $centos_file ; done
+  if [[ -e /etc/centos-release ]]; then
```
better: `-f`

shlomi-noach commented on conf/orchestrator-sample-sqlite.conf.json:
```diff
@@ -0,0 +1,120 @@
+  "GraphiteAddr": "",
+  "GraphitePath": "",
+  "GraphiteConvertHostnameDotsToUnderscores": true
+}
```
:+1:

shlomi-noach commented on go/agent/agent_dao.go:
```diff
@@ -693,7 +693,7 @@ func executeSeed(seedId int64, targetHostname string, sourceHostname string) err
 
 	seedFromLogicalVolume := sourceAgent.LogicalVolumes[0]
-	seedStateId, _ = submitSeedStateEntry(seedId, fmt.Sprintf("Mounting logical volume: %s", seedFromLogicalVolume.Path), "")
+	seedStateId, _ = submitSeedStateEntry(seedId, fmt.Sprintf("%s Mounting logical volume: %s", sourceHostname, seedFromLogicalVolume.Path), "")
```
:+1:

shlomi-noach commented on go/agent/agent_dao.go:
```diff
@@ -722,8 +722,8 @@ func executeSeed(seedId int64, targetHostname string, sourceHostname string) err
-	seedStateId, _ = submitSeedStateEntry(seedId, fmt.Sprintf("Waiting some time for %s to start listening for incoming data", targetHostname), "")
-	time.Sleep(2 * time.Second)
+	seedStateId, _ = submitSeedStateEntry(seedId, fmt.Sprintf("Waiting %d seconds for %s to start listening for incoming data", config.Config.SeedWaitSecondsBeforeSend, targetHostname), "")
+	time.Sleep(time.Duration(config.Config.SeedWaitSecondsBeforeSend) * time.Second)
```
:+1:

shlomi-noach commented on go/db/generate_base.go:
```diff
@@ -138,7 +138,7 @@ var generateSQLBase = []string{
 		  last_seen timestamp NULL DEFAULT NULL,
 		  mysql_port smallint(5) unsigned DEFAULT NULL,
-		  count_mysql_snapshots smallint(5) unsigned NOT NULL,
+		  count_mysql_snapshots smallint(5) unsigned NOT NULL DEFAULT 0,
```
🚫 Must change: existing SQL must never ever change. The most we can do is add a modification in `generate_patches.go`. However, modifying a column is not supported by `sqlite` so this change must unfortunately go away.

The value is implicitly `0` anyway, what is the concern?

Slach replied:
oh. ok. sorry i will rollback this change

Slach replied:
yes concern is 
count_mysql_snapshots not used when try insert into ```host_agent``` table
and sql query is failed when agent submitting

shlomi-noach reviewed (changes requested):
Thank you for the PR! and I'm sorry for the slow response; I have had some busy days and will have yet more busy days ahead of me.
One critical change, and some questions.

Slach committed:
return rsync shared folder back
rollback generate_base.go
after
https://github.com/github/orchestrator/pull/445#discussion_r177999552

Slach committed:
Merge branch 'master' of github.com:github/orchestrator into reanimate_agent

Slach committed:
fix agent

shlomi-noach commented on build.sh:
```diff
@@ -134,12 +134,14 @@ function package() {
+        # n CentOD 6 box: we only want the rpms for CentOS6
+        # Add "-centos6" to the file name.
+        ls ${TOPDIR:-?}/*.rpm | while read f; do centos_file=$(echo $f | sed -r -e "s/^(.*)-${RELEASE_VERSION}(.*)/\1-centos6-${RELEASE_VERSION}\2/g") ; mv $f $centos_file ; done
+      fi
```
:+1:

shlomi-noach commented on vagrant/admin-build.sh:
```diff
@@ -1,23 +1,34 @@
 
 # Discover instances
-/usr/bin/orchestrator -c discover -i localhost
+/usr/bin/orchestrator --verbose --debug --stack -c redeploy-internal-db
```
What happened that the above became necessary? Current logic should make it safe to run `orchestrator` without specifying `redeploy-internal-db`.

Slach replied:
ohh, sorry, i multiple run ```vagrant up``` and add redeploy-internal-db to try fix errors when using sqlite backend
i revert this changes right now

Slach committed:
remove redeploy-internal-db

Slach commented:
@shlomi-noach would this PR ready for merge? i think i resolve any recomendation

Slach committed:
Merge branch 'master' of github.com:github/orchestrator into reanimate_agent

Slach committed:
Merge branch 'master' into reanimate_agent

Slach committed:
Merge branch 'reanimate_agent' of github.com:Slach/orchestrator into reanimate_agent

Commit:
Merge branch 'master' into reanimate_agent

Commit:
Merge branch 'master' into reanimate_agent

Slach commented:
thanks!

shlomi-noach commented:
Thank you!
